modules/ding_mobilesearch/ding_mobilesearch.module

```diff
@@ -0,0 +1,504 @@
+              'attr' => array(),
+            );
+            $mapping['fields'][$machine_name . '__' . $key]['value'][$ikey] = $value;
+          }
+        }
+        break;
+    }
+  }
+
+  $module_enabled = system_list('module_enabled');
+  if (isset($module_enabled['opening_hours'])) {
+    $result = db_select('opening_hours', 'oh')
+      ->fields('oh')
+      ->condition('oh.nid', $node->nid)
+      ->execute();
+    foreach ($result as $row) {
+      $mapping['fields']['opening_hours'][] = (array) $row;
+    }
+  }
+
+  $tag_field = NULL;
+  // Content type specific fields.
+  switch ($node->type) {
+    case 'ding_event':
+      $items = field_get_items('node', $node, 'field_ding_event_date');
+      if (is_array($items) && !empty($items)) {
+        $item = reset($items);
+        $mapping['fields']['field_ding_event_date'] = array(
+          'name' => 'Event date',
+          'value' => array(
+            'from' => $item['value'],
+            'to' => $item['value2'],
+          ),
+          'attr' => array(),
+        );
+      }
+      $tag_field = 'field_ding_event_tags';
+      break;
+
+    case 'ding_news':
+      $tag_field = 'field_ding_news_tags';
+      break;
+
+    case 'ding_page':
+      $tag_field = 'field_ding_page_tags';
+      break;
+  }
+
+  if ($tag_field) {
+    $items = field_get_items('node', $node, $tag_field);
+    if (is_array($items) && !empty($items)) {
+      $mapping['fields'][$tag_field] = array(
+        'name' => 'Tags',
+        'value' => array(),
+        'attr' => array(),
+      );
+      foreach ($items as $item) {
+        $mapping['fields'][$tag_field]['value'][] = $item['name'];
+      }
+    }
+  }
+
+  return $mapping;
+}
+
+/**
+ * Implements hook_action_info().
+ *
+ * Registers custom VBO actions as Drupal actions.
+ */
+function ding_mobilesearch_action_info() {
+  return array(
+    'views_bulk_operations_push_mongo_rest' => array(
+      'type' => 'entity',
+      'label' => t('Push to Mongo'),
+      'configurable' => FALSE,
+      'triggers' => array('any'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_ding_install_tasks().
+ */
+function ding_mobilesearch_ding_install_tasks() {
+  // module_load_include('inc', 'ding_mobilesearch', 'includes/admin');
+  return array(
+    'ding_mobilesearch_settings' => array(
+      'display_name' => st('Mobilesearch mapping settings'),
+      'type' => 'form',
+      'file' => drupal_get_path('module', 'ding_mobilesearch') . '/includes/admin.inc',
+    ),
+    'ding_mobilesearch_plugin_settings' => array(
+      'display_name' => st('Mobilesearch plugins settings'),
+      'type' => 'form',
+      'file' => drupal_get_path('module', 'ding_mobilesearch') . '/includes/admin.inc',
+    ),
+  );
+}
+
+/**
+ * Implements hook_flag_default_flags().
+ */
+function ding_mobilesearch_flag_default_flags() {
+  $types = node_type_get_names();
+  $flags = array();
+  // Exported flag: "Push to Mongo".
+  $flags['push_to_mongo'] = array(
```

Comment:
@cableman The Service Provider ("OpenPlatform") uses this node-client: https://github.com/DBCDK/dbc-node-ddbcontent-client. This is a client that communicated with a REST API used in "MobileSearch" and "Filmstriben" projects. It holds events, news and libraries in a MongoDB.